compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -898,12 +898,16 @@ abstract class GenJSCode extends plugins.PluginComponent
 
     private def genClassInterfaces(sym: Symbol, forJSClass: Boolean)(
         implicit pos: Position): List[js.Ident] = {
+
+      val blacklist: Set[Symbol] =
+        if (forJSClass) Set(DynamicClass, SerializableClass) // #3118, #3252
```

Comment:
Ok with me, but you might want to consider making this set more static for speed purposes.